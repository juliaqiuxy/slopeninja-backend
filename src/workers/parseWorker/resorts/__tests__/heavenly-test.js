import fs from 'fs';
import {
  parseHeavenlySnow,
  parseHeavenlyLiftCounts,
  parseHeavenlyLifts,
  parseHeavenlyTrails,
  parseHeavenlyTrailCounts,
} from '../heavenly';
import { createHtmlParser } from '../../parserFactory';

test('fetches Heavenly snow data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/heavenly-weather.html.input`);

  const resortData = await createHtmlParser('snow', parseHeavenlySnow)(htmlText);
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: 0,
      snowDepthBase: null,
      snowDepthSummit: 25,
    },
  });
});

test('fetches all null for nonexisting snow values', async () => {
  const resortData = await createHtmlParser('snow', parseHeavenlySnow)('<html></html>');
  expect(resortData).toEqual({
    snow: {
      status: null,
      weatherIcon: null,
      temperature: null,
      baseCondition: null,
      newSnow: null,
      snowDepthBase: null,
      snowDepthSummit: null,
    },
  });
});

test('fetches Heavenly lifts data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/heavenly-lifts.html.input`);
  const resortData = await createHtmlParser('liftCounts', parseHeavenlyLiftCounts)(htmlText);
  expect(resortData).toEqual({
    liftCounts: {
      total: 27,
      open: 16,
    },
  });
});

test('fetches all null for nonexisting lift values', async () => {
  const resortData = await createHtmlParser('liftCounts', parseHeavenlyLiftCounts)('<html></html>');
  expect(resortData).toEqual({
    liftCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Heavenly trails data correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/heavenly-lifts.html.input`);
  const resortData = await createHtmlParser('trailCounts', parseHeavenlyTrailCounts)(htmlText);
  expect(resortData).toEqual({
    trailCounts: {
      total: 110,
      open: 59,
    },
  });
});

test('fetches all null for nonexisting trails values', async () => {
  const resortData = await createHtmlParser('trailCounts', parseHeavenlyTrailCounts)('<html></html>');
  expect(resortData).toEqual({
    trailCounts: {
      total: null,
      open: null,
    },
  });
});

test('fetches Heavenly lift list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/heavenly-lifts.html.input`);
  const resortData = await createHtmlParser(
    'lifts',
    parseHeavenlyLifts,
  )(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser(
    'lifts',
    parseHeavenlyLifts,
  )('<html></html>');
  expect(resortData).toMatchObject({ lifts: [] });
});

test('fetches Heavenly trail list correctly', async () => {
  const htmlText = fs.readFileSync(`${__dirname}/__fixtures__/heavenly-lifts.html.input`);
  const resortData = await createHtmlParser(
    'trails',
    parseHeavenlyTrails,
    // removeBackSlashes, // REMOVE FROM INDEX
  )(htmlText);
  expect(resortData).toMatchSnapshot();
});

test('fetches all null for nonexisting lift list values', async () => {
  const resortData = await createHtmlParser(
    'trails',
    parseHeavenlyTrails,
  )('<html></html>');
  expect(resortData).toMatchObject({ trails: [] });
});
