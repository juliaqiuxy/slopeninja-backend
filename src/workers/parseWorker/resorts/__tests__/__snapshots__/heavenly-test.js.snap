// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fetches Heavenly lift list correctly 1`] = `
Object {
  "lifts": Array [
    Object {
      "category": null,
      "name": "Aerial Tram",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Bear Cave Carpet",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Big Easy",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Boulder",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Boulder Carpet",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Canyon",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Comet",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Dipper",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "DMZ Carpet",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Enchanted Carpet",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "First Ride",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Galaxy",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Gondola",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Groove",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Gunbarrel",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Mott",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "North Bowl",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Olympic",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Patsy's",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Pioneer Mitey Mite",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Powderbowl",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Red Fir Mitey Mite",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "Sky",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Stagecoach",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Tamarack",
      "status": "open",
    },
    Object {
      "category": null,
      "name": "Tubing Lift",
      "status": "closed",
    },
    Object {
      "category": null,
      "name": "World Cup",
      "status": "closed",
    },
  ],
}
`;

exports[`fetches Heavenly trail list correctly 1`] = `
Object {
  "trails": Array [
    Object {
      "category": "California",
      "level": "expert",
      "name": "Advanced Round-a-Bout",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "California Trail Lower",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Canyonland",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Double Down Lower",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Double Down Upper",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "advanced",
      "name": "East Bowl",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Ellie's Lower",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Ellie's Swing",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Ellie's Upper",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Express Line",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Fall Line",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "advanced",
      "name": "Gunbarrel Lower",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "advanced",
      "name": "Gunbarrel Upper",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "High Five",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Liz's Lower",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Liz's Upper",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "beginner",
      "name": "Maggie's",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Mombo",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Mombo Meadows",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "beginner",
      "name": "Patsy's",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Pinnacles Left",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Pinnacles Right",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Pistol",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "beginner",
      "name": "Poma Trail",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Powder Line",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Powderbowl Run",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Powderbowl Woods",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Ridge Bowl",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Ridge Run",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Ridgeway",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Round-a-Bout",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Round-a-Bout Swing",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Ski Way Glades",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Sky Canyon",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Sky Chute",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Skyline trail",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Stein's Way",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Swing Trail",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "advanced",
      "name": "The Face",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "Upper Mombo",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "Waterfall",
      "status": "open",
    },
    Object {
      "category": "California",
      "level": "expert",
      "name": "West Bowl",
      "status": "closed",
    },
    Object {
      "category": "California",
      "level": "intermediate",
      "name": "World Cup",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "$100 Saddle",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Aries",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Aries Woods",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Big Dipper Lower",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Big Dipper Upper",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Bohemian Grove",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Bonanza",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "beginner",
      "name": "Boulder Bowl",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Boulder Chute",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Cloud Nine",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Comet",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Comstock Lower",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Comstock Upper",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Cosmic Wave",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Crossover",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Dipper Bowl",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Dipper Line",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Dipper Woods",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "beginner",
      "name": "Easiest Way",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Emily's Run",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Galaxy",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Galaxy Line",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Jack's",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "advanced",
      "name": "Killebrew Canyon",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Little Dipper",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Lower Dipper Return",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Meteor",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Milky Way",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Milky Way Bowl",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Mineshaft Lower",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Mineshaft Upper",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "advanced",
      "name": "Mott Canyon",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Nevada Lower Trail",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Nevada Trail Upper",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Nevada Woods",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "North Bowl",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Nova",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Olympic Down Hill Lower",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Olympic Down Hill Mid",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Olympic Down Hill Upper",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Orion",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Orion's Belt",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Outlaw",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Pepi's",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Perimeter",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Ponderosa",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Sand Dunes",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Stagecoach Lower",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Stagecoach Middle",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Stagecoach Return",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Stagecoach Upper",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Stagecoach Woods",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "expert",
      "name": "Stageline",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "The Pines",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Upper Dipper Return",
      "status": "open",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Upper North Bowl",
      "status": "closed",
    },
    Object {
      "category": "Nevada",
      "level": "intermediate",
      "name": "Upper Perimeter",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "intermediate",
      "name": "49er",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "beginner",
      "name": "Big Easy",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "beginner",
      "name": "Black Bear Hollow",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "intermediate",
      "name": "California Trail Upper",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "beginner",
      "name": "Easy Street",
      "status": "closed",
    },
    Object {
      "category": "Top of Gondola",
      "level": "intermediate",
      "name": "Sam's Dream",
      "status": "closed",
    },
    Object {
      "category": "Top of Gondola",
      "level": "intermediate",
      "name": "Silver Spur",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "intermediate",
      "name": "Tamarack Return",
      "status": "open",
    },
    Object {
      "category": "Top of Gondola",
      "level": "intermediate",
      "name": "VonSchmidt's",
      "status": "open",
    },
  ],
}
`;
